edgecast/resources/cps/certificate.go

```diff
@@ -279,6 +349,126 @@ func ExpandDomains(attr interface{}) ([]*models.DomainCreateUpdate, error) {
 	}
 }
 
+// ExpandNotifSettings converts the Terraform representation of organization
+// into the EmailNotification API Model.
+func ExpandNotifSettings(
+	attr interface{},
+) ([]*models.EmailNotification, []error) {
+	if attr == nil {
+		return make([]*models.EmailNotification, 0), nil
+	}
+
+	tfSet, ok := attr.(*schema.Set)
+	if !ok {
+		return nil, []error{errors.New("error parsing notification settings")}
+	}
+
+	if tfSet == nil {
+		return make([]*models.EmailNotification, 0), nil
+	}
+
+	maps := tfSet.List()
+
+	// Empty map
+	if len(maps) == 0 {
+		return make([]*models.EmailNotification, 0), nil
+	}
+
+	errs := make([]error, 0)
+	emailNotifs := make([]*models.EmailNotification, 0)
+
+	for ix, v := range maps {
+		m, ok := v.(map[string]any)
+		if !ok {
+			errs = append(
+				errs,
+				fmt.Errorf("error parsing notification_setting[%d]", ix))
+
+			continue
+		}
+
+		emailNotif := &models.EmailNotification{}
+
+		if notifType, ok := GetStringFromMap(m, "notification_type"); ok {
+			emailNotif.NotificationType = notifType
+		} else {
+			err := fmt.Errorf(
+				"error parsing notification_setting[%d].notification_type",
+				ix)
+			errs = append(errs, err)
+		}
+
+		if enabled, ok := GetBoolFromMap(m, "enabled"); ok {
+			emailNotif.Enabled = enabled
+		} else {
+			err := fmt.Errorf(
+				"error parsing notification_setting[%d].enabled",
+				ix)
+			errs = append(errs, err)
+		}
+
+		emails, err := helper.ConvertTFCollectionToStrings(m["emails"])
+		if err == nil {
+			emailNotif.Emails = emails
+		} else {
+			err := fmt.Errorf(
+				"error parsing notification_setting[%d].emails: %w",
+				ix,
+				err)
+			errs = append(errs, err)
+		}
+
+		emailNotifs = append(emailNotifs, emailNotif)
+	}
+
+	return emailNotifs, errs
+}
+
+func GetStringFromMap(m map[string]any, key string) (string, bool) {
```

Comment:
Are you planning to add unit tests for these functions in the future (or are they moving out of certificate.go into helper?)